### ISSUE TITLE
Mouse pointer is flickering + High CPU usage on file write 

### ISSUE THREAD

**Issue by zeoint** [manga-download/haruneko#446]
Mouse pointer starts flickering when the progress bar is full and stops after some time.

This is happening because of images in the downloaded folder being swapped or something. This did not happen in hakuneko.

---

https://github.com/manga-download/haruneko/assets/127277950/41ae74db-1505-45df-938f-da26d40b57e2

**Comment by Sheepux** [comment]
Yup i'm aware of this one and didn't find the cause of the issue :\

**Comment by MikeZeDev** [comment]
Its quite easy to reproduce. Just downloaded a 400 pages chapter( testing requested website multporn.net ). Pages are downloaded quite fast but are not in destination folder until quite some time. While they are moved to the destination folder i got this mouse flickering thing.

**Comment by Kharlitoz21** [comment]
What happens is that while the "download" bar is loading, no chapter is saved in the folder. It’s only at the end, when the bar finishes, that it starts saving all the chapter's images at once. This causes the cursor to freeze momentarily, and my CPU usage spikes to 100%.

**Comment by Kharlitoz21** [comment]
My solution would be for each image to be saved one by one during the download, so that by the time it finishes, all the images are already in the folder, instead of being saved all at once at the end.

**Comment by 27136773** [comment]
Fils are moved to the target location after all images of the chapter are downloaded successfully.
This is by design to avoid storing images in the target directory in case the chapter download fails midway.
